src/formatDate.js

```diff
@@ -8,7 +8,31 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const separatorFrom = fromFormat[3];
+  const separatorTo = toFormat[3];
+  const dateParts = date.split(separatorFrom);
+  const dateMap = {};
+
+  fromFormat.slice(0, 3).forEach((part, index) => {
+    dateMap[part] = dateParts[index];
+  });
+
+  const convertToYYYY = (yy) => (parseInt(yy, 10) < 30 ? `20${yy}` : `19${yy}`);
+  const convertToYY = (yyyy) => yyyy.slice(-2);
```

Comment:
Similarly to the previous comment, you may want to ensure `yyyy` is exactly four digits long before slicing the last two digits. This is to maintain consistency and guard against unexpected input lengths.